### PR TITLE
Extrinsic bug with undefined value

### PR DESCRIPTION
**Pull request by Stefie**
Addresses the `Method.js:104 Method: cannot decode value 'undefined' of type undefined` Bug in https://github.com/polkadot-js/apps/issues/1160 

Onload on https://polkadot.js.org/apps/#/toolbox
Screenshot 2019-05-15 at 14 41 31

### CODE REVIEW

**Comment by amaurymartiny** [comment]
I'm okay to merge this, but tbh I think it won't change anything:

When value is undefined, we pass in `{}` to decodeExtrinsic

https://github.com/polkadot-js/api/blob/d9ecedba2545ac4ad0626c27e1e4cc8f86ae650c/packages/types/src/type/Extrinsic.ts#L41

so `value` never be undefined inside decodeExtrinsic's args afaict

**Comment by Stefie** [comment]
> I'm okay to merge this, but tbh I think it won't change anything:

@amaurymartiny at least it makes this bug polkadot-js/apps#1160 in the toolbox app disappear  :)

File: packages/types/src/primitive/Method.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default class Method extends Struct implements IMethod {
       args: new Uint8Array(),
       argsDef: {},
       meta: new FunctionMetadata(),
-      callIndex: new Uint8Array([255, 255])
+      callIndex: new Uint8Array()
```
Comment (jacogr) [inline packages/types/src/primitive/Method.ts]: Throw on line 125, line 88 becomes -

`private static decodeMethod (value: DecodedMethod | Uint8Array | string = new Uint8Array(), _meta?: FunctionMetadata): DecodedMethod {`

**Comment by jacogr** [comment]
Still think apps needs to be fixed to not pass crap info.

**Comment by jacogr** [comment]
@amaurymartiny Yes, you are 100% right - so think it is looking at the wrong thing. (Don't understand how that makes the issue disappear)

**Comment by Stefie** [comment]
> Don't understand how that makes the issue disappear)

Good to hear, because me neither! Good thing is that I fixed two other bugs while trying to investigate that...

File: packages/types/src/primitive/Method.ts (hunk 1 above)
Comment (Stefie) [reply packages/types/src/primitive/Method.ts]: I changed that (if I understood you correctly)

Comment (jacogr) [reply packages/types/src/primitive/Method.ts]: 100%, line 125 should throw, everything below that should be removed. (No return - at that point, we give up)

**Comment by Stefie** [comment]
The changes in `Method.ts` were not directly related to the bug, but I changed it to throwing an error.

**Pull request merged**